[4.0] Build fails with a JDK 8

Although the build claims to require [only a JDK 8](https://github.com/neo4j-contrib/neo4j-jdbc/blob/03886f20976c6ab1f2379dab6cec2d6b4ea96fcf/pom.xml#L78), it fails with such:

```shell
$ mvn clean package -DskipTests
[...]
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.1:testCompile (default-testCompile) on project neo4j-jdbc-bolt: Compilation failure
[ERROR] /Users/fbiville/workspace/neo4j-jdbc/neo4j-jdbc-bolt/src/test/java/org/neo4j/jdbc/bolt/BoltNeo4jSetParameterDateIT.java:[7,25] cannot access org.neo4j.graphdb.Result
[ERROR]   bad class file: /Users/fbiville/.m2/repository/org/neo4j/neo4j-graphdb-api/4.0.0/neo4j-graphdb-api-4.0.0.jar(org/neo4j/graphdb/Result.class)
[ERROR]     class file has wrong version 55.0, should be 52.0
[ERROR]     Please remove or make sure it appears in the correct subdirectory of the classpath.
```

No issue with a JDK 11.

--- a/.gitignore
+++ b/.gitignore
@@ -32,3 +32,5 @@ target/
 #Neo4j
 **/neo4j-home/*
 
+# jenv
+/.java-version

--- a/neo4j-jdbc-bolt/src/test/java/org/neo4j/TestcontainersCausalCluster.java
+++ b/neo4j-jdbc-bolt/src/test/java/org/neo4j/TestcontainersCausalCluster.java
@@ -178,22 +178,6 @@ public class TestcontainersCausalCluster {
                 .orElseThrow(() -> new IllegalStateException("No sidecar as entrypoint into the cluster available."));
     }
 
-    public URI getURIByType(ClusterInstanceType instanceType) {
-        return this.sidecars.stream()
-                .filter(instance -> instance.getLabels().getOrDefault("memberType", "").equals(instanceType.toString()))
-                .findAny()
-                .map(instance -> String.format("bolt+routing://%s:%d", instance.getContainerIpAddress(),
-                        instance.getMappedPort(DEFAULT_BOLT_PORT)))
-                .map(uri -> {
-                    try {
-                        return new URI(uri);
-                    } catch (URISyntaxException e) {
-                        throw new RuntimeException(e);
-                    }
-                })
-                .orElseThrow(() -> new IllegalStateException("No sidecar as entrypoint into the cluster available."));
-    }
-
     public URI getAllMembersURI() {
         try {
             return new URI("bolt+routing://" + this.sidecars.stream()

--- a/neo4j-jdbc-bolt/src/test/java/org/neo4j/TestcontainersCausalCluster.java
+++ b/neo4j-jdbc-bolt/src/test/java/org/neo4j/TestcontainersCausalCluster.java
@@ -1,6 +1,5 @@
 package org.neo4j;
 
-import org.jetbrains.annotations.NotNull;
 import org.neo4j.driver.AuthTokens;
 import org.neo4j.driver.Driver;
 import org.neo4j.driver.GraphDatabase;
@@ -85,7 +84,6 @@ public class TestcontainersCausalCluster {
         return new TestcontainersCausalCluster(members, sidecars.values().stream().collect(toList()));
     }
 
-    @NotNull
     private static List<Neo4jContainer> getClusterMembers(int numberOfCoreMembers,
                                                                    ClusterInstanceType instanceType,
                                                                    Map<String, GenericContainer> sidecars,
@@ -104,7 +102,6 @@ public class TestcontainersCausalCluster {
                 .collect(toList());
     }
 
-    @NotNull
     private static Map<String, GenericContainer> createSidecars(int numOfMembers, Network network, ClusterInstanceType instanceType) {
         return iterateMembers(numOfMembers, instanceType)
                 .collect(toMap(

--- a/pom.xml
+++ b/pom.xml
@@ -257,6 +257,7 @@
 				<configuration>
 					<source>${java.version}</source>
 					<target>${java.version}</target>
+					<forceJavacCompilerUse>true</forceJavacCompilerUse>
 				</configuration>
 			</plugin>
 			<plugin>

--- a/pom.xml
+++ b/pom.xml
@@ -75,7 +75,7 @@
 	<properties>
 		<!-- General -->
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-		<java.version>1.8</java.version>
+		<java.version>11</java.version>
 
 		<!-- Deps version -->
 		<powermock.version>2.0.4</powermock.version>

--- a/pom.xml
+++ b/pom.xml
@@ -85,7 +85,7 @@
 		<neo4j.version>4.0.0</neo4j.version>
 		<httpclient.version>4.5.3</httpclient.version>
 		<jackson.version>2.11.0</jackson.version>
-		<testcontainers.version>1.12.3</testcontainers.version>
+		<testcontainers.version>1.15.0</testcontainers.version>
 
 		<!-- Plugin version -->
 		<maven.plugin.compiler>3.6.1</maven.plugin.compiler>

